subjects/java/checkpoints/single-linkedlist/README.md

```diff
@@ -0,0 +1,119 @@
+## Single Linked List
+
+### Instructions
+
+Create a single linked list data structure that implements the following methods:
+
+- `at(int index)`: to access an element by its index, if the index is out of bound return -1.
+- `add(int value)`: to add an element at the end of the list.
+- `remove(int index)`: to remove an element by its index.
+- `size()`: to get theh size of the list.
+
+Define these methods in an interface called `LinkedList`, and implement this interface in a class that you will create. Additionally, add a private method `next(Node node)` in your class that will be used to traversing the list in the other methods. This method should print the message "Go to next node" each time it is called.
```

Comment:
To avoid confusion, I would add that the message "Go to the next node" should be followed by a new line

Reply:
done